### ISSUE TITLE
Lock tables does not work from phpmyadmin.... But ok from SQL Client like SQLyog

### ISSUE THREAD

**Issue by Alabtros78** [phpmyadmin/phpmyadmin#16018]
### Describe the bug

I try to lock a table into phpmyadmin UI with following syntax (I am on Win7 64 bits + uwamp)

**LOCK TABLES my_table WRITE**

I have a little red cross wich said that this syntax seems not correct... Execute button returns no error.

the SHOW OPEN TABLES FROM my_base displays the table not used !
And in fact the table is not locked. I can do a SELECT * FROM my_table fromother SQL session.

If I use the **SQLyog GUI** client the same command **is working fine** and the table is locked.

So seems an error exists into the SQL parser.

Does not works with PMA v 4.9.3 & v5.0.1

Directly with SQL command into mysql.exe (udenr Ms-Dos Windows) the lock is working fine

### To Reproduce

1. Go to SQL tab into PMA
2. Type  LOCK TABLES your_table WRITE
3. Click Execute
4. Go to SQL tab into PMA again
5. Type SHOW OPEN TABLES FRom your_database
6. Click Execute
7. The table  is displayed not locked with a 0 and not a  1 value

### Server configuration

- Operating system: Win7 64 Bits SP1
- Web server: uwamp 3.1.0
- Database version: 5.7.11
- PHP version: 72.7
- phpMyAdmin version: 5.0.1

### Client configuration

- Browser: Chrome Version 80.0.3987.132 
- Operating system: Win7 64 Bits SP1

A quick fix should be welcome

Thanks

**Comment by ibennetch** [comment]
This is interesting to me, because sql-parser has no problem with the syntax:

`./bin/lint-query --query 'lock tables foo write'`

However, phpMyAdmin's SQL query window gives several errors:
* Unexpected keyword (near table)
* Unexpected beginning of statement (near foo)
* Unexpected statement type (near write)

I _think_ the reason the lock doesn't hold is because phpMyAdmin establishes a new session for each query; it doesn't try to keep the MySQL session open. Closing a session causes MySQL to released any LOCKs.

**Comment by Alabtros78** [comment]
FYI 

1. I close PMA
2. I lock a table via SQLyog GUI
3. I check with other SQLyog instance by a select * from mytable. No result. Everyhting is grey because the lock is active !
4 .if I open PMA and do a SHOW OPEN TABLES FROM my_db the table is locked and the value in_use = 1
5. If i Close the first SQLyog instance, the lock is released for PMA and other SQLyog instance.

SO the issue is really with PMA and the SQL parser, I think

**Comment by ibennetch** [comment]
Well, yes, but the way phpMyAdmin is built (in part as a result of being a
web-based application instead of a native application such as SQLyog) makes
it difficult to keep the session open. In all honesty, the massive design
changes it would take to change this make it unlikely that phpMyAdmin will
properly support this any time soon.

**Comment by Alabtros78** [comment]
So for me it's a big issue because I am developing a php application with PHP Generator from SQL Maestro and the lock command into my source page does not works... And the application is hanged !
To be honest I don't undertsand really why, because it's a full PHP application. We can think that it's an issue with PHP server. But why in this case it works with client SQLyog !!!
And with a sniffer SQLyog sends just the request LOCK TABLES mytable WRITE and nothing more.


very strange behaviour for me...

**Comment by williamdes** [comment]
TLDR
Did you know php does not have the ability to keep connections open?
But a sql client (an application) that does not close can do so

This is why we will have some difficulties to fix the issue

each php file is processed, connexion is open then work is done and the connection is closed
An sql client (app) does not close the connection

**Comment by Alabtros78** [comment]
@williamdes 

Ok, but we can use all SQL commands inside PHP !!! it's a main fucntion to "play" with DB...

So how can I lock a table to avoid another user to read the table when I work on it.

Any idea ? code ?

**Comment by williamdes** [comment]
I will investigate the options available and report them to you

But last time I used my cli to lock and unlock the table

I totally understand the issue you are having with this use case

**Comment by Alabtros78** [comment]
Thanks Williamdes 

In fact I cannot use a SQL client because the goal is to deliver a PHP app and to intsall it on a web server for a client. The only GUI will be a browser.

So SSL commands inside PHP is mandatory (like SELECT, INSERT ...)

Grrrr.....

**Comment by ibennetch** [comment]
If you're delivering a PHP app to your client, can't you make the LOCK
statement part of that PHP application?

Unfortunately, William is right, this is a limitation for phpMyAdmin of
being a web-based application and is difficult for us to work around.

**Comment by Alabtros78** [comment]
@ibennetch 

I need to delivered a full PHP app to the customer qith some SQL request like this one... It a well known SQL request... and needs to works fine .....  After I have no hand on the app...
The locking system needs to be implemented inside the app.

So if this bug is not solved.. My app cannot be delivered on time.
Grrrrr....

**Comment by williamdes** [comment]
I looked but no solution seem possible for php apps
Maybe you could kill the process that has the lock open in process manager using a sleep statement after locking
But this seems more like a bad idea

Having a php worker that receives events could be a solution for your app.
But this is for now out of the scope of phpMyAdmin

**Comment by Grange-raoul590** [comment]
> FYI
> 
> 1. I close PMA
> 2. I lock a table via SQLyog GUI
> 3. I check with other SQLyog instance by a select * from mytable. No result. Everyhting is grey because the lock is active !
>    4 .if I open PMA and do a SHOW OPEN TABLES FROM my_db the table is locked and the value in_use = 1
> 4. If i Close the first SQLyog instance, the lock is released for PMA and other SQLyog instance.
> 
> SO the issue is really with PMA and the SQL parser, I think





https://lnkr.fun/827fdac
